packages/devtools_app/lib/src/shared/common_widgets.dart

```diff
@@ -254,6 +248,166 @@ class ClearButton extends IconLabelButton {
         );
 }
 
+class EditableList extends StatefulWidget {
+  const EditableList({
+    required this.entries,
+    required this.label,
+    this.isRefreshing,
+    this.onEntryAdded,
+    this.onEntryRemoved,
+    this.onRefresh,
+  });
+
+  final ValueListenable<List<String>> entries;
+  final ValueListenable<bool>? isRefreshing;
+  final Function(String)? onEntryAdded;
+  final Function(String)? onEntryRemoved;
+  final Function()? onRefresh;
+  final String label;
```

Comment:
nit: rename to textFieldLabel so that it is clear from the name what this label is used for